sdks/java/io/amazon-web-services/src/main/java/org/apache/beam/sdk/io/aws/options/AwsModule.java

```diff
@@ -234,4 +239,23 @@ public SSEAwsKeyManagementParams deserialize(JsonParser parser, DeserializationC
       return new SSEAwsKeyManagementParams(awsKmsKeyId);
     }
   }
+
+  @JsonAutoDetect(
+    fieldVisibility = Visibility.NONE,
+    getterVisibility = Visibility.NONE,
+    setterVisibility = Visibility.NONE
+  )
+  interface ClientConfigurationMixin {
```

Comment:
This is a pretty interesting way to do this.

Reply:
I started down the path of a interface mixin that just @JsonIgnore'ed the properties that would not serialize property (look at my commit history). But then there were messed up defaults since it was setting those properties to null rather than their default value. I considered what it would take to make this mixin properly serialize all those classes, and I saw too low of an ROI. I considered just adding discrete string properties to the AwsSettings class, but that did not feel like the right design either. I settled on this as the least bad solution.